logic/src/commonMain/kotlin/com/wire/kalium/logic/data/client/ClientMapper.kt

```diff
@@ -0,0 +1,85 @@
+package com.wire.kalium.logic.data.client
+
+import com.wire.kalium.logic.data.location.LocationMapper
+import com.wire.kalium.logic.data.prekey.PreyKeyMapper
+import com.wire.kalium.network.api.user.client.Capabilities
+import com.wire.kalium.network.api.user.client.ClientCapabilityDTO
+import com.wire.kalium.network.api.user.client.ClientTypeDTO
+import com.wire.kalium.network.api.user.client.DeviceTypeDTO
+import com.wire.kalium.network.api.user.client.RegisterClientRequest
+import com.wire.kalium.network.api.user.client.RegisterClientResponse
+
+class ClientMapper(
+    private val preyKeyMapper: PreyKeyMapper,
+    private val locationMapper: LocationMapper
+) {
+
+    fun toRegisterClientRequest(param: RegisterClientParam): RegisterClientRequest = RegisterClientRequest(
+        password = param.password,
+        lastKey = preyKeyMapper.toPreKeyDTO(param.lastKey),
+        label = param.label,
+        deviceType = toDeviceTypeDTO(param.deviceType),
+        type = toClientTypeDTO(param.type),
+        capabilities = param.capabilities?.let { it.map { toClientCapabilityDTO(it) } } ?: run { null },
+        model = param.model,
+        preKeyDTOS = param.preKeys.map { preyKeyMapper.toPreKeyDTO(it) },
+    )
+
+    fun fromRegisterClientResponse(response: RegisterClientResponse): Client = Client(
```

Comment:
would this make sense and extension function on RegisterClientResponse?

Reply:
If we want to take logic as the only entry point for external consumers of Kalium, we should be quite vigilant on the visibility methods we qualify/declare our network objects